src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -249,6 +256,36 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
       assert(v2 > 0)
       assert(v1 + v2 == indexLogEntry.content.fileInfos.size)
     }
+
+    // Modify the file again.
+    val sourcePath2 = new Path(spark.read.parquet(nonPartitionedDataPath).inputFiles.last)
+    fs.copyToLocalFile(sourcePath2, deletedFile)
+
+    new RefreshAppendAction(
+      spark,
+      IndexLogManagerFactoryImpl.create(indexPath),
+      IndexDataManagerFactoryImpl.create(indexPath))
+      .run()
+
+    {
+      // Check non-empty deletedFiles after RefreshAppendAction.
+      val indexLogEntry = getLatestStableLog(indexConfig.indexName)
+      assert(indexLogEntry.deletedFiles.size == 1)
+      assert(indexLogEntry.appendedFiles.isEmpty)
+      assert(logManager(systemPath, indexConfig.indexName).getLatestId().get == 7)
+      val files = indexLogEntry.relations.head.data.properties.content.files
+      assert(files.exists(_.equals(deletedFile)))
+      val v1 = indexLogEntry.content.fileInfos
+        .count(_.name.contains(s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=1"))
```

Comment:
How about the following local function?
```scala
def getIndexFilesCount(entry: IndexLogEntry, version: Int) = {
  entry.content.fileInfos
    .count(_.name.contains(s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=$version"))
  // you can assert the count is > 0 if you want since it's local function just for this test.
}
```